specs/span-batches.md

```diff
@@ -292,6 +282,8 @@ Span-batch rules, in validation order:
         is past `inclusion_block_number` because of the following invariant.
       - Invariant: the epoch-num in the batch is always less than the inclusion block number,
         if and only if the L1 epoch hash is correct.
+    - `start_epoch_num < prev_l2_block.l1_origin.number` -> `drop`:
+      epoch number cannot be older than the origin of parent block
```

Comment:
Original spec when we did not allow overlapping span batch:
```
    - `start_epoch_num < epoch.number` -> `drop`: must have been duplicate batch,
      we may be past this L1 block in the safe L2 chain. If a span-batch overlaps with older information,
      it is dropped, since partially valid span-batches are not accepted.
```

This condition cannot be checked by using current block hash. I cannot recall why we removed this check at https://github.com/testinprod-io/optimism/pull/23. 

I think adding this rule is correct. We must check that `start_epoch_num` is valid, based on its corresponding `prev_l2_block`.

Reply:
> One question: why this condition is necessary? Assuming the current block's hash is legit, doesn't it prove that the previous L2 block's L1 Origin number is legit since it contains a cryptographical commit of the previous block?

Can you define what current block you are mentioning?

Lets say all hash checks are passed: `batch.parent_check == prev_l2_block.hash[:20]` and `batch.l1_origin_check` is correct.

How can we validate `start_epoch_num` based on this info? `start_epoch_num` is calculated from `batch.l1_origin_num` and `origin_bits`. 

Condition `start_epoch_num < prev_l2_block.l1_origin.number` cannot be checked using hash check.

Reply:
Invariant: **L1 origin cannot change by more than one L1 block per L2 block.**.

If overlapping: the L1 origin derived using `originBits` and `batch. l1_origin_num` must be matched with safe chain. If we do not add this check, `Overlapped blocks checks` will eventually catch the invalid `start_epoch_num`.

If entire span batch is not overlapping and not future: this means `prev_l2_block` equals to `safe_head`. We must enforce the invariant. Therefore `start_epoch_num >= prev_l2_block.l1_origin.number` <=> `start_epoch_num >= safe_head.l1_origin.number` must be still satisfied.

By ANDing two conditions, we must add the condition(drop when `start_epoch_num < prev_l2_block.l1_origin.number`).

Reply:
gud 👍